### ISSUE TITLE
Unit.Visualization.Python.GenerateXdmf - AttributeError: module 'spectre.Informer' has no attribute 'unit_test_path'

### ISSUE THREAD

**Issue by wthrowe** [sxs-collaboration/spectre#1824]
I have confirmed that this occurs in a clean build.

```
    Start 1030: Unit.Visualization.Python.GenerateXdmf
1/1 Test #1030: Unit.Visualization.Python.GenerateXdmf ...***Failed    0.25 sec
/usr/lib64/python3.6/site-packages/h5py/__init__.py:34: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
test_generate_xdmf (__main__.TestGenerateXdmf) ... ERROR

======================================================================
ERROR: test_generate_xdmf (__main__.TestGenerateXdmf)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/wthrowe/spectre/tests/Unit/Visualization/Python/Test_GenerateXdmf.py", line 14, in test_generate_xdmf
    spectre_informer.unit_test_path(), 'IO',
AttributeError: module 'spectre.Informer' has no attribute 'unit_test_path'
```

LibraryVersions.txt:
```
CMake Version: 3.14.6
SpECTRE Version: 0.0.0
SpECTRE branch: develop
SpECTRE hash: 7f123150d94add8bd6734b58b8c7c2819f8b7e3e
Hostname: ungoliant
Host system: Linux-4.14.83-gentoo
Host system version: 4.14.83-gentoo
Host system processor: x86_64
CMAKE_CXX_FLAGS: -fPIC -fdiagnostics-color=always -ferror-limit=1 -ftemplate-backtrace-limit=0 -march=native -fPIC
CMAKE_CXX_LINK_FLAGS: 
CMAKE_CXX_FLAGS_DEBUG: -g
CMAKE_CXX_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_C_FLAGS: -fPIC -march=native -fPIC
CMAKE_C_LINK_FLAGS: 
CMAKE_C_FLAGS_DEBUG: -g
CMAKE_C_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_Fortran_FLAGS: -march=native
CMAKE_Fortran_LINK_FLAGS: 
CMAKE_Fortran_FLAGS_DEBUG: -g
CMAKE_Fortran_FLAGS_RELEASE: -O3 -DNDEBUG -O3
CMAKE_EXE_LINKER_FLAGS: -march=native -no-pie
CMAKE_BUILD_TYPE: Debug
CMAKE_CXX_COMPILER: /home/wthrowe/spectre-clang-debug-64/filter-clang++-7
CMAKE_CXX_COMPILER_VERSION: 7.1.0
CMAKE_C_COMPILER: /home/wthrowe/spectre-clang-debug-64/filter-clang-7
CMAKE_C_COMPILER_VERSION: 7.1.0
CMAKE_Fortran_COMPILER: /usr/bin/gfortran
CMAKE_Fortran_COMPILER_VERSION: 6.4.0
Python Version: 3.6.5
Charm Version:  6.8.0
Charm SMP:  YES
BLAS_LIBRARIES:  /usr/lib64/blas/reference/libblas.so
Blaze Version:  3.2
Boost Version:  1.65.0
Catch Version:  2.9.1
Google Benchmark Found
GSL Version:  2.5
HDF5 Version:  1.10.5
jemalloc Version:  ..
LIBXSMM Version:  1.8.1
LAPACK_LIBRARIES:  /usr/lib64/lapack/reference/liblapack.so;/usr/lib64/blas/reference/libblas.so
NumPy Version:  1.14.5
SciPy Version:  1.1.0
```

**Comment by nilsvu** [comment]
Did this failure also occur in the CI, i.e. in the Docker container? I see you’re using Py3 so it’s probably not in the container?

**Comment by wthrowe** [comment]
I have not seen this in CI, but I have not looked.  This is not in the container.  I get basically the same error with python 2:
```
          Start 1030: Unit.Visualization.Python.GenerateXdmf
1030/1030 Test #1030: Unit.Visualization.Python.GenerateXdmf ......................................***Failed    0.17 sec
/usr/lib64/python2.7/site-packages/h5py/__init__.py:34: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
test_generate_xdmf (__main__.TestGenerateXdmf) ... ERROR

======================================================================
ERROR: test_generate_xdmf (__main__.TestGenerateXdmf)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/wthrowe/spectre/tests/Unit/Visualization/Python/Test_GenerateXdmf.py", line 14, in test_generate_xdmf
    spectre_informer.unit_test_path(), 'IO',
AttributeError: 'module' object has no attribute 'unit_test_path'
```
LibraryVersions.txt: (I don't know why "Python Version" is blank.  That's probably another bug.  It's 2.7.15.)
```
CMake Version: 3.14.6
SpECTRE Version: 0.0.0
SpECTRE branch: develop
SpECTRE hash: 7f123150d94add8bd6734b58b8c7c2819f8b7e3e
Hostname: ungoliant
Host system: Linux-4.14.83-gentoo
Host system version: 4.14.83-gentoo
Host system processor: x86_64
CMAKE_CXX_FLAGS: -fPIC -fdiagnostics-color=always -ferror-limit=1 -ftemplate-backtrace-limit=0 -march=native -Werror -fPIC
CMAKE_CXX_LINK_FLAGS: 
CMAKE_CXX_FLAGS_DEBUG: -g
CMAKE_CXX_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_C_FLAGS: -fPIC -march=native -Werror -fPIC
CMAKE_C_LINK_FLAGS: 
CMAKE_C_FLAGS_DEBUG: -g
CMAKE_C_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_Fortran_FLAGS: -march=native
CMAKE_Fortran_LINK_FLAGS: 
CMAKE_Fortran_FLAGS_DEBUG: -g
CMAKE_Fortran_FLAGS_RELEASE: -O3 -DNDEBUG -O3
CMAKE_EXE_LINKER_FLAGS: -march=native -Werror -no-pie
CMAKE_BUILD_TYPE: Release
CMAKE_CXX_COMPILER: /home/wthrowe/tmp/spectre-test-all/clang-7-opt-64/filter-clang++-7
CMAKE_CXX_COMPILER_VERSION: 7.1.0
CMAKE_C_COMPILER: /home/wthrowe/tmp/spectre-test-all/clang-7-opt-64/filter-clang-7
CMAKE_C_COMPILER_VERSION: 7.1.0
CMAKE_Fortran_COMPILER: /usr/bin/gfortran
CMAKE_Fortran_COMPILER_VERSION: 6.4.0
Python Version: 
Charm Version:  6.8.0
Charm SMP:  YES
BLAS_LIBRARIES:  /usr/lib64/blas/reference/libblas.so
Blaze Version:  3.2
Boost Version:  1.65.0
Catch Version:  2.9.1
Google Benchmark Found
GSL Version:  2.5
HDF5 Version:  1.10.5
jemalloc Version:  ..
LIBXSMM Version:  1.8.1
LAPACK_LIBRARIES:  /usr/lib64/lapack/reference/liblapack.so;/usr/lib64/blas/reference/libblas.so
NumPy Version:  1.14.5
SciPy Version:  1.1.0
```

**Comment by nilsvu** [comment]
Did you `make test-executables` before running `ctest`?

**Comment by nilsvu** [comment]
Ah I think I understand the issue. You probably don't have `BUILD_PYTHON_BINDINGS` enabled, but since `GenerateXdmf.py` doesn't need bindings it still being built (=symlinked) and tested. But its _test_ requires the Pybindings to be built, so it fails. I'll have to think about what the best solution for this is...

**Comment by wthrowe** [comment]
Ah, that sounds right.  I have the bindings disabled because of #1518.  (Although I would still probably do some test builds with them disabled even without that.)

**Comment by nilsvu** [comment]
Yes, I fix #1518 in #1782 (which needs a refactor before getting merged). And I plan on distributing {Py2, Py3, no bindings} configurations over the CI builds.

**Comment by nilsvu** [comment]
To fix this issue, I think these are the options:
- Require Pybindings to be built for Python code that doesn't need them by itself, but that has tests that need them.
- Disable tests that require Pybindings if they are not built, even if the code they are testing _is_ built.
- Require that tests for Python code that doesn't need Pybindings _also_ can't use Pybindings. In the code of `GenerateXdmf` this means we need to get the path to a sample volume data file without calling `unit_test_path`..

**Comment by nilsvu** [comment]
@wthrowe can you confirm that this issue is resolved? We went with option 2 in my list above. I will fix #1518 in #1782.

**Comment by wthrowe** [comment]
Yes, I think it was fixed in #1836.
